config/reader.go

```diff
@@ -294,10 +294,11 @@ func readFromCloud(
 		to.Cloud.LocationSecrets = locationSecrets
 		to.Cloud.TLSCertificate = tls.certificate
 		to.Cloud.TLSPrivateKey = tls.privateKey
+		to.Cloud.PrimaryOrgID = primaryOrgID
+		to.Cloud.LocationID = locationID
```

Comment:
I had to make this change for my new tests to pass, which would indicate that we are caching these new fields but not making them available in our loaded config. However, this should lead to a server failure since we validate against these fields.

Reply:
:facepalm: wow, these are indeed missing. `GetCloudMetadata` has been returning nothing this whole time.